### ISSUE TITLE
git module changes repo URL and returns "changed": false when run with --check

### ISSUE THREAD

**Issue by famousgarkin** [ansible/ansible-modules-core#4006]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

git module
##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.1.0.0
  config file = 
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

```
[defaults]
nocows = yes
vault_password_file = .vault-password
allow_world_readable_tmpfiles = yes
```
##### OS / ENVIRONMENT
- control host: OS X 10.11.5
- managed host: Debian 7, Debian 8
##### SUMMARY

git module changes repo URL and returns "changed": false when run with --check
##### STEPS TO REPRODUCE

``` yaml
# git-clone.yml

- hosts: localhost
  gather_facts: no
  tasks:
  - local_action: git repo={{url}} dest=ansible version=v1.0
```

```
$ ansible-playbook git-clone.yml -e url=https://github.com/ansible/ansible.git -v

PLAY [localhost] ***************************************************************

TASK [git] *********************************************************************
changed: [localhost -> localhost] => {"after": "6a64e9f0248d402aa60faf33bc720acbcd56e50b", "before": null, "changed": true}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0   

$ git -C ansible remote -v
origin  https://github.com/ansible/ansible.git (fetch)
origin  https://github.com/ansible/ansible.git (push)

$ ansible-playbook git-clone.yml -e url=https://github.com/famousgarkin/ansible.git -v --check

PLAY [localhost] ***************************************************************

TASK [git] *********************************************************************
ok: [localhost -> localhost] => {"after": "6a64e9f0248d402aa60faf33bc720acbcd56e50b", "before": "6a64e9f0248d402aa60faf33bc720acbcd56e50b", "changed": false}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0   

$ git -C ansible remote -v
origin  https://github.com/famousgarkin/ansible.git (fetch)
origin  https://github.com/famousgarkin/ansible.git (push)
```
##### EXPECTED RESULTS
- returns `"changed": true`
- repo URL is not changed
##### ACTUAL RESULTS
- returns `"changed": false`
- repo URL is changed

**Comment by robinro** [comment]
One needs to put some check for check_mode around https://github.com/ansible/ansible-modules-core/blob/devel/source_control/git.py#L817

@famousgarkin maybe this is something you can try to submit a PR for yourself? Some quick fix should be doable in a couple of lines.

Also it would be nice to have a test-case for this, potentially by using https://github.com/ansible/ansible/pull/16056

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)
